auth/client-api.md

```diff
@@ -0,0 +1,22 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
```

Comment:
should we add following methods:
`requestAuthentication()` - for app that creates a pairing and sends `wc_authRequest`, 
`pair(`) ?
`ping()` ?